src/app/optimizer/optimizer.component.ts

```diff
@@ -15,12 +15,14 @@ import {ResultsTableComponent} from './results-table/results-table.component';
   providers: [OptimizerDataService, HTTP_PROVIDERS]
 })
 export class OptimizerComponent {
-  optimalAllocs: Object;
+  optimalAllocs;
   sharpeRatio: Number;
+  trailingDecimals = 4;
```

Comment:
This value will never change, so we should define as `const trailingDecimals = 4;`